src/axom/quest/MarchingCubesAlgo.hpp

```diff
@@ -0,0 +1,257 @@
+
+// Conduit includes
+#include "conduit_node.hpp"
+
+// C++ includes
+#include <string>
+
+namespace axom
+{
+namespace quest
+{
+/*!
+ * \@brief Class implementing marching cubes algorithm on a single
+ * structured mesh domain.
+ *
+ * Implementation is for 2D (marching squares) and 3D.
+ *
+ * \sa MarchingCubesAlgo
+ */
+class MarchingCubesAlgo1
+{
+public:
+  /*!
+   * \brief Constructor for applying algorithm in multi-level mesh context.
+   *
+   * \param [in] dom Blueprint single-domain mesh containing scalar field.
+   * \param [in] coordsetName Name of blueprint point coordinate set.
+   * \param [in] maskField Cell-based std::int32_t mask field.  If provided,
+   *             cells where this field evaluates to false are skipped.
+   *
+   * Some data from \a dom may be cached by the constructor.
+   * Any change to it after the constructor leads to undefined behavior.
+   * See set_domain(const conduit::Node &) for requirements on \a dom.
+   *
+   * The mesh coordinates should be contiguous.  See
+   * conduit::blueprint::is_contiguous().  In the future, this
+   * requirement may be relaxed, possibly at the cost of a
+   * transformation and storage of the temporary contiguous layout.
+   */
+  MarchingCubesAlgo1(const conduit::Node &dom,
+                     const std::string &coordsetName,
+                     const std::string &maskfield);
+
+  //! @brief Spatial dimension of domain.
+  int dimension() const { return _ndim; }
+
+  /*!
+    @brief Set the field containing the nodal function.
+    \param [in] fcnField Name of node-based scalar function values.
+  */
+  void set_function_field(const std::string &fcnField);
+
+  /*!
+    @brief Set the output surface mesh object.
+  */
+  void set_output_mesh(axom::mint::Mesh *surfaceMesh);
+
+  /*!
+    @brief Save the originating cell index in output mesh.
+
+    For each contour mesh cell generated, the 1D index of the
+    generating computational mesh cell is set in the \a cellIdField of
+    the contour mesh.  If the field doesn't exist, it will be created.
+    To disable, set \a outputCellIds to empty.
+  */
+  void set_cell_id_field(const std::string &cellIdField)
+  {
+    _cellIdField = cellIdField;
+  }
+
+  /*!
+   * \brief Computes the iso-surface.
+   *
+   * \param [in] contourVal iso-contour value
+   *
+   * Compute iso-surface using the marching cubes algorithm.
+   */
+  void compute_iso_surface(double contourVal = 0.0);
+
+private:
+  /*!
+    \brief Computational mesh as a conduit::Node.
+
+    Plan for supporting blueprint mesh in sidre is to shallow-copy the
+    mesh to a conduit::Node on the heap, and point _dom to it.
+  */
+  const conduit::Node *_dom;
+  int _ndim;
+  axom::Array<axom::IndexType> _cShape;  //!< @brief Shape of cell-centered data arrays in _dom.
+  axom::Array<axom::IndexType> _logicalOrigin;  //!< @brief First domain cell in each direction.
+
+  const std::string _coordsetPath;
+  std::string _fcnPath;
+  std::string _maskPath;
```

Comment:
Style:
Axom's style is to have member variables begin with `m_`. Please update.

Reply:
👍

Reply:
Done.